pkg/converter/converter_test.go

```diff
@@ -0,0 +1,191 @@
+metadata:
+  name: default
+spec:`,
+			expectedOutput: `apiVersion: sailoperator.io/v1
+kind: Istio
+metadata:
+  name: default
+spec:
+  version: %s
+  namespace: %s`,
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			g := NewWithT(t)
+			err := saveYamlToFile(tc.input, istioFile)
+			g.Expect(err).To(Succeed(), "failed to write YAML file")
+
+			if tc.name == "mandatory-arguments-only" {
+				g.Expect(executeConfigConverter(istioFile, "", "", istioVersion)).To(Succeed(), "error in execution of ./configuration-converter.sh")
+			} else {
+				g.Expect(executeConfigConverter(istioFile, sailFile, controlPlaneNamespace,
+					istioVersion)).To(Succeed(), "error in execution of ./configuration-converter.sh")
+			}
+			tc.expectedOutput = fmt.Sprintf(tc.expectedOutput, istioVersion, controlPlaneNamespace)
+			isConversionValidated, err := validateYamlContent(tc.expectedOutput, sailFile)
+			g.Expect(err).To(Succeed(), "Can not open file to compare")
+			g.Expect(isConversionValidated).To(BeTrue(), "Converted content is not as expected")
+		})
+	}
+}
+
+// saveYamlToFile writes the given YAML content to given file
+func saveYamlToFile(input, istioFile string) error {
+	if err := os.WriteFile(istioFile, []byte(input), 0o644); err != nil {
+		return fmt.Errorf("failed to write YAML file: %w", err)
+	}
+	return nil
+}
+
+func executeConfigConverter(input, output, controlPlaneNamespace, istioVersion string) error {
+	converter := filepath.Join(project.RootDir, "tools", "configuration-converter.sh")
+	args := []string{converter, input}
+
+	if output != "" {
+		args = append(args, output)
+	}
+
+	if controlPlaneNamespace != "" {
+		args = append(args, "-n", controlPlaneNamespace)
+	}
+
+	args = append(args, "-v", istioVersion)
+
+	cmd := strings.Join(args, " ")
+	_, err := shell.ExecuteCommand(cmd)
+	if err != nil {
+		return fmt.Errorf("error in execution of %s %s %s -n %s -v %s: %w", converter, input, output, controlPlaneNamespace, istioVersion, err)
+	}
+	return nil
+}
+
+// compareYamlContent checks if the provided YAML content matches the content of converted sail operator config
+func validateYamlContent(expectedOutput, sailFile string) (bool, error) {
+	// Write the input YAML string to a temporary file
+	tmpFile := filepath.Join(project.RootDir, "tools", "temp.yaml")
+	err := os.WriteFile(tmpFile, []byte(expectedOutput), 0o644)
+	if err != nil {
+		return false, fmt.Errorf("failed to write temporary YAML file: %w", err)
+	}
+	defer os.Remove(tmpFile)
+
+	// The command will check if the files are equal, ignoring order
+	cmd := fmt.Sprintf("diff <(yq -P 'sort_keys(..)' -o=props %s) <(yq -P 'sort_keys(..)' -o=props %s)", sailFile, tmpFile)
+	output, err := shell.ExecuteCommand(cmd)
+	if err != nil {
+		return false, fmt.Errorf("error executing yq comparison: %w", err)
+	}
+	if output != "" {
+		return false, nil
+	}
+	// If no output from yq, the files are considered equal
+	return true, nil
+}
```

Comment:
There's a better way to do this. You can use the `go-cmp` library to compare two objects (perhaps even strings?). We already do this in other tests; for example: https://github.com/istio-ecosystem/sail-operator/blob/8326acf08ecc509fc93eea29f581582903076e41/controllers/istio/istio_controller_test.go#L544